oauthenticator/auth0.py

```diff
@@ -6,6 +6,7 @@
 The following environment variables may be used for configuration:
 
     AUTH0_SUBDOMAIN - The subdomain for your Auth0 account
+    AUTH0_CUSTOM_DOMAIN - The custom domain for your Auth0 account
```

Comment:
The original design in #207 had this called `AUTH0_DOMAIN`, which I agree is better. That this can be a custom domain is more of a marketing term. It is not really "custom" from the perspective of `oauthenticator`. It is just a regular domain name. You could put `mycompany.auth0.com` in this variable and it would work perfectly, even though it is not a custom domain by Auth0's definition.